integrations/java/openhouse-java-runtime/src/main/java/com/linkedin/openhouse/javaclient/OpenHouseTableOperations.java

```diff
@@ -276,21 +276,22 @@ static Mono<GetTableResponseBody> handleCreateUpdateHttpError(
       return Mono.error(
           new CommitFailedException(
               casted, casted.getStatusCode().value() + " , " + casted.getResponseBodyAsString()));
-    } else if (e instanceof WebClientResponseException.GatewayTimeout
-        | e instanceof WebClientResponseException.ServiceUnavailable) {
-      return Mono.error(new CommitStateUnknownException(e));
     } else if (e instanceof WebClientResponseException.BadRequest) {
       WebClientResponseException casted = (WebClientResponseException) e;
       return Mono.error(
           new BadRequestException(
               casted, casted.getStatusCode().value() + " , " + casted.getResponseBodyAsString()));
```

Comment:
Was wondering if we can do something like this. Print casted, statusCode and ResponseBody

Reply:
Sure, I see your point. [BadRequestException](https://github.com/apache/iceberg/blob/00f46ac0960237324e69401e98caa79c70407da5/api/src/main/java/org/apache/iceberg/exceptions/BadRequestException.java#L31) does not extract the message from throwable whereas [CommitStateUnknown](https://github.com/apache/iceberg/blob/00f46ac0960237324e69401e98caa79c70407da5/api/src/main/java/org/apache/iceberg/exceptions/CommitStateUnknownException.java#L36) does it.  So extracting the status code might be redundant. Wdyt?

Reply:
Does both status code and response body get logged as part of exception stacktrace? If so we are good with the observable we need.

Reply:
Modified to capture responseBody since statusCode is already present